lib/libc/riscv/SYS.h

```diff
@@ -42,22 +42,46 @@
 	li	t0, SYS_ ## name;				\
 	ecall
 
+#ifndef __CHERI_PURE_CAPABILITY__
+#define _GET_LOCAL_FNPTR(outreg, function)	\
+	lla outreg, _C_LABEL(function)
+#define _GET_FNPTR(outreg, function)	\
+	la outreg, _C_LABEL(function)
+#define _CALL_FNPTR(reg)	jalr cra, reg
```

Comment:
s/cra/ra/?  You can also just use `jalr reg` as a pseudo-instruction where `ra` is implied.